Lib/collections/__init__.py

```diff
@@ -1010,9 +1011,8 @@ def __len__(self):
         return len(set().union(*self.maps))     # reuses stored hash values if possible
 
     def __iter__(self):
-        d = {}
-        for mapping in reversed(self.maps):
-            d.update(dict.fromkeys(mapping))    # reuses stored hash values if possible
+        lazy_views = (key_view.keys() for key_view in reversed(self.maps))
+        d = dict.fromkeys(_chain.from_iterable(lazy_views))
```

Comment:
This doesn't do what you think it does.  The chain extracts the keys from the views and then passes those through to `dict.fromkeys()`.  That defeats the goal of retaining the precomputed hash values stored in the underlying mapping.

The `dict.fromkeys()` class method can use the precomputed hash values ONLY if the input is an exact dict or exact set.  An iterable of keys will not suffice.

Also, there isn't any "laziness" at all in these two lines.  You make a generator expression and then consume it immediately.  It is not much different from converting `sum([10, 20, 30])` to sum(iter([10, 20, 30]))` and calling it "lazy".